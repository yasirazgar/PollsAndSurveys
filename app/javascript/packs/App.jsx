// Run this example by adding <%= javascript_pack_tag 'hello_react' %> to the head of your layout file,
// like app/views/layouts/application.html.erb. All it does is render <div>Hello React</div> at the bottom
// of the page.

import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import Home from './Home.jsx'
import reducers from '../reducers'
import loadTranslations from '../i18n'
import { buildTranslations } from '../actions'


const translations = loadTranslations()
// const createStoreWithMiddleware = applyMiddleware()(createStore(reducers));
const store = createStore(reducers, applyMiddleware(thunk));
store.dispatch(buildTranslations(translations))

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Home />
      </Provider>
    )
  }
}

document.addEventListener('DOMContentLoaded', () => {
  ReactDOM.render(<App />, document.querySelector('.app'))
})

export default App