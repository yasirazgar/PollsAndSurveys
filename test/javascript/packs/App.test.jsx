import React from 'react';
import { shallow, mount } from 'enzyme';
import App from 'packs/App';

describe("App", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it("sets state sideDrawerOpen to true on clicking sideDrawer toggle-button", () => {
    const wrapper = mount(<App />);

    wrapper.find('.toggle-button').first().simulate('click');

    expect(wrapper.state('sideDrawerOpen')).toEqual(true);
  });
});
