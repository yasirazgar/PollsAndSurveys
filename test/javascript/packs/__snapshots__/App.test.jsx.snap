// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Home />
</Provider>
`;
